future-apps/payroll/contracts/Payroll.sol

```diff
@@ -533,19 +550,20 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
      * @dev Send the requested amount of salary to the employee
      * @param _employeeId Employee's identifier
      * @param _requestedAmount Amount of owed salary requested (must be less than or equal to total owed salary). Using `0` will request all available salary
-     * @return True if something has been paid
      */
-    function _payday(uint256 _employeeId, uint256 _requestedAmount) internal returns (bool somethingPaid) {
+    function _payday(uint256 _employeeId, uint256 _requestedAmount) internal {
         Employee storage employee = employees[_employeeId];
 
         // Compute amount to be payed
-        uint256 owedAmount = _getOwedSalary(_employeeId);
-        require(owedAmount > 0, ERROR_NOTHING_PAID);
-        require(owedAmount >= _requestedAmount, ERROR_INVALID_REQUESTED_AMOUNT);
-        uint256 payingAmount = _requestedAmount > 0 ? _requestedAmount : owedAmount;
+        uint256 currentOwedSalary = _getOwedSalary(_employeeId);
+        uint256 totalOwedSalary = currentOwedSalary + employee.accruedSalary;
+        if (totalOwedSalary < currentOwedSalary) totalOwedSalary = MAX_UINT256; // cap to max int if it overflows
```

Comment:
Let's use a new block with braces for this, and make the comment a bit louder :)